appinventor/components/src/com/google/appinventor/components/runtime/util/WebRTCNativeMgr.java

```diff
@@ -392,29 +424,34 @@ private void Poller() {
     }
   }
 
-  private void sendRendezvous(JSONObject data) {
-    try {
-      data.put("first", first);
-      data.put("webrtc", true);
-      data.put("key", rCode + "-r");
-      if (first) {
-        first = false;
-        data.put("apiversion", SdkLevel.getLevel());
-      }
-      HttpClient client = new DefaultHttpClient();
-      HttpPost post = new HttpPost("http://" + rendezvousServer2 + "/rendezvous2/");
-      try {
-        if (DEBUG) {
-          Log.d(LOG_TAG, "About to send = " + data.toString());
+  private void sendRendezvous(final JSONObject data) {
+    AsynchUtil.runAsynchronously(new Runnable() {
```

Comment:
Does it make sense in the high latency case to do something similar to what we did with CloudDB where we keep a queue of ICE candidates and if the latency is high and they bunch together we can send them in a single request?

Reply:
We need to do the sending to the Rendezvous server asynchronously so we don't block the native thread that is otherwise negotiating WebRTC. We could build a queue instead, but it would be heard to know when it is time to send it. There is no "I'm all done with ice candidates" that is given to us. So we would just have to wait an arbitrary amount, which probably isn't a good idea. Note: The spec has an option to send a null candidate as an indicator of completion, but the library we are using doesn't do that.